app/models/user.rb

```diff
@@ -0,0 +1,82 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                         :bigint           not null, primary key
+#  email                      :string(255)      not null
+#  password_hash              :string(255)
+#  token                      :uuid
+#  first_name                 :string(255)
+#  last_name                  :string(255)
+#  phone_number               :string(255)
+#  inserted_at                :datetime         not null
+#  updated_at                 :datetime         not null
+#  email_verification_token   :string(255)
+#  email_verified_at          :datetime
+#  role                       :string(255)      default("user"), not null
+#  password_reset_token       :uuid
+#  password_reset_expires_at  :datetime
+#  finalized                  :boolean          default(TRUE), not null
+#  avatar_key                 :uuid
+#  avatar_extension           :string(255)
+#  display                    :boolean          default(TRUE), not null
+#  terms_of_use               :datetime
+#  privacy_guidelines         :datetime
+#  agency_id                  :bigint
+#  last_active                :datetime
+#  status                     :string(255)
+#  active_session             :boolean          default(FALSE)
+#  renewal_request            :string(255)
+#  jwt_token                  :text
+#  recertification_expired_at :datetime
+#
+class User < ApplicationRecord
+  has_many :challenges
+  has_many :challenge_managers
+  has_many :challenge_manager_challenges, through: :challenge_managers, source: :challenge
+  has_many :members
+  has_many :supporting_documents, class_name: 'Document'
+  has_many :submissions, foreign_key: :submitter_id
+  has_many :managed_submissions, class_name: 'Submission', foreign_key: :manager_id
+  has_many :submission_documents, class_name: 'Submissions::Document'
+  has_many :message_context_statuses
+
+  # Virtual Attributes
+  attr_accessor :email_confirmation, :password, :password_confirmation
```

Comment:
the docs for the attribute method say it doesn't need to be backed by the DB. I'm not sure that we need the password/password_confirmation attributes since users shouldn't be able to reset their login.gov password from the app AFAIK.
![Screenshot 2024-07-22 at 5 18 35 PM](https://github.com/user-attachments/assets/4ff2c0f5-1497-4bbc-9bde-8b8a6cefd26c)